sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/partition_processor.py

```diff
@@ -17,16 +17,16 @@ class CloseReason(Enum):
     EVENTHUB_EXCEPTION = 2  # Exception happens during receiving events
 
 
-class PartitionProcessor(ABC):
+class PartitionProcessor(Protocol):
```

Comment:
@johanste I am fairly new to MyPy, I may have some gaps in my understanding. From what I know, `Protocol` is useful to move runtime (type or structure) checks to something that MyPy can report on. But that's only useful to the people that are using MyPy, i.e. the developers of a library, more or less. To me this suggests that `Protocol` is most useful for things that end-users are *not* expected to subclass themselves. If end-users are expected to make their own subclasses, they can't be assumed to be using MyPy, so then you will still need all the explicit runtime checks (see all the `hasattr` checks above). In which case, isn't an `ABC` is better, so that the library devs don't have to perform these checks explicitly everywhere?

Reply:
When I changed from ABC to Protocol I was also thinking using a subclass looks more natural to me.  Or did I use the Protocol in a wrong way? @johanste

Reply:
Changed back to use ABC instead of Protocol